libafl/src/mutators/tuneable.rs

```diff
@@ -200,17 +199,32 @@ where
             return ret;
         }
 
-        // We will sample using the mutation probabilities.
-        // Doing this outside of the original if branch to make the borrow checker happy.
-        #[allow(clippy::cast_precision_loss)]
-        let coin = state.rand_mut().next() as f32 / u64::MAX as f32;
-        let metadata = TuneableScheduledMutatorMetadata::get_mut(state).unwrap();
-        metadata
-            .mutation_probabilities_cumulative
-            .iter()
-            .position(|i| *i >= coin)
-            .unwrap()
-            .into()
+        if !metadata.mutation_probabilities_cumulative.is_empty() {
+            // We will sample using the mutation probabilities.
+            // Doing this outside of the original if branch to make the borrow checker happy.
+            #[allow(clippy::cast_precision_loss)]
+            let coin = state.rand_mut().next() as f32 / u64::MAX as f32;
+            debug_assert!(coin <= 1.0_f32);
+
+            let metadata = TuneableScheduledMutatorMetadata::get_mut(state).unwrap();
+            debug_assert_eq!(
```

Comment:
I wanted to move this check to when we set the probabilities, but that method doesn't have access to the tuneable mutations. Any idea on how to do that? (and if that would be preferred).

Reply:
Don't think it's possible without major restructuring